builtin/bisect--helper.c

```diff
@@ -574,23 +579,30 @@ static int bisect_start(struct bisect_terms *terms, int no_checkout,
 	 * Check for one bad and then some good revisions
 	 */
 	for (i = 0; i < argc; i++) {
-		if (!strcmp(argv[i], "--")) {
+		char *arg = dequote_arg(argv[i]);
+
+		if (!strcmp(arg, "--")) {
 			has_double_dash = 1;
+			free(arg);
 			break;
 		}
+		free(arg);
 	}
 
 	for (i = 0; i < argc; i++) {
-		const char *arg = argv[i];
+		char *dequoted = dequote_arg(argv[i]);
+		const char *arg = dequoted;
```

Comment:
I don't think you need the `dequoted` variable anymore. I think you can use `arg` everywhere.

Previously it was needed because args that were dequoted were allocated new memory while args that didn't need dequoting weren't. But now we always allocate memory in `dequote_arg()`.

Reply:
If we use `char *arg = dequote_arg(argv[i]);`

We get the following warning at `skip_prefix()`.
```
builtin/bisect--helper.c:606:47: warning: passing 'char **' to parameter of type
      'const char **' discards qualifiers in nested pointer types
      [-Wincompatible-pointer-types-discards-qualifiers]
                } else if (skip_prefix(arg, "--term-good=", &arg) ||
```

Reply:
Did you try `const char *arg = dequote_arg(argv[i]);`?